src/GraphQL.Client/GraphQLHttpClientOptions.cs

```diff
@@ -50,12 +50,17 @@ public class GraphQLHttpClientOptions
             Task.FromResult(request is GraphQLHttpRequest graphQLHttpRequest ? graphQLHttpRequest : new GraphQLHttpRequest(request));
 
         /// <summary>
-        /// This callback is called after successfully establishing a websocket connection but before any regular request is made. 
+        /// Delegate to determine if GraphQL response may be properly deserialized into <see cref="GraphQLResponse{T}"/>.
+        /// </summary>
+        public Func<HttpResponseMessage, bool> IsValidResponseToDeserialize { get; set; } = r => r.IsSuccessStatusCode;
```

Comment:
Shouldn't we accept status 400 by default? As far as I can see that's whats returned when query validation fails in [graphql-dotnet/server](https://github.com/graphql-dotnet/server/blob/83c8ccaf12d259e2f3b65c9ce1f2982e95eb5e3b/src/Transports.AspNetCore/GraphQLHttpMiddleware.cs#L281)...

I think we should aim for default compatibility between those two repos...

```suggestion
        public Func<HttpResponseMessage, bool> IsValidResponseToDeserialize { get; set; } = r => r.IsSuccessStatusCode || r.StatusCode == HttpStatusCode.BadRequest;
```

Reply:
I'm fine to accept 400 codes but it is some sort of semi-breaking changes and suites more if published in v5 some time ago.